app/src/main/java/io/bluetrace/opentrace/onboarding/HowItWorksFragment.kt

```diff
@@ -4,16 +4,37 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import com.google.firebase.auth.FirebaseAuth
 import io.bluetrace.opentrace.R
+import io.bluetrace.opentrace.logging.CentralLog
 
 class HowItWorksFragment : OnboardingFragmentInterface() {
 
+    private var TAG: String = "HowItWorksFragment"
+
     override fun getButtonText(): String = "Continue"
 
     override fun becomesVisible() {}
 
     override fun onButtonClick(view: View) {
-        val onboardActivity = context as OnboardingActivity
+        authenticate()
+    }
+
+    private fun authenticate() {
+        val firebaseAuth = FirebaseAuth.getInstance()
+        firebaseAuth.signInAnonymously()
+            .addOnCompleteListener(requireActivity()) { task ->
+                if (task.isSuccessful) {
+                    CentralLog.i(TAG, "Sign in success")
```

Comment:
we don't care if this doesn't work?

Reply:
I am not sure what to do if we can't sign in.
I mean the app will work but the functions will be failing.
Any suggestions on what should we do when fail?

Reply:
Maybe just add a `//TODO Error handling needed` just in case we have time 👍

Reply:
Done